Include/internal/pycore_pystate.h

```diff
@@ -226,6 +226,7 @@ typedef struct pyruntimestate {
            If that becomes a problem later then we can adjust, e.g. by
            using a Python int. */
         int64_t next_id;
+        int finalizing;
```

Comment:
It may make more sense to use a flag indicating that the runtime is ready to add (sub)interpreters.  That would have the same meaning as `finalizing` in your case (though an inverted value), but is more broadly useful (including the option to explicitly disable subinterpreters, or pause their availability, using the flag).

In my order of preference (I'm 50/50 on the first two):

```suggestion
        int available;
```

```suggestion
        int ready;
```

```suggestion
        int supported;
```

```suggestion
        int can_have_subinterpreters;
```

I'd expect the flag to be initialized to 0 (false) in `_PyRuntime_Initialize()` and set to 1 at the same place that `_PyRuntime.initialized` is set to 1 (in `pyinit_main()`).  Then you'd set it in the same places that you are setting `finalizing` below (but with inverted values, of course).

Reply:
I've gone for Nick's suggestion of `allow_new`, if you're happy with that?

Reply:
I guess it's a bit vague.  "new" what?  Honestly, the flag represents more than just the availability of subinterpreters.  So I'd probably feel more comfortable with "ready" (or "available"), which would mean "the full capability of the C-API is available" (including subinterpreters).

@ncoghlan, what do you think?

Reply:
There are relevant comments below not included in this thread - in short I don't have strong feelings here, so I'll leave it as 'allow_new' unless someone (Eric?) wants me to change it again!